src/waypoint.jsx

```diff
@@ -333,13 +352,18 @@ export default class Waypoint extends React.Component {
    * @return {Object}
    */
   render() {
+    if (this.props.children) {
+      return <div ref={this.refElement}>{this.props.children}</div>;
```

Comment:
Ah true, cloning could make this a little more convenient.

If we do that, it would be good to mention in the documentation that functional components cannot be passed to Waypoint in this way, since they cannot have refs because they don't have instances. If there is an easy enough way to detect this, it might be worth making Waypoint give a helpful error message in dev.